experimental/dds/tree/src/SharedTree.ts

```diff
@@ -1311,8 +1333,8 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 	 */
 	public revert(editId: EditId): EditId | undefined {
 		const index = this.edits.getIndexOfId(editId);
-		const edit = this.edits.getEditInSessionAtIndex(index);
-		const before = this.logViewer.getRevisionViewInSession(index);
+		const edit = assertNotUndefined(this.edits.tryGetEditAtIndex(index), 'edit not found');
```

Comment:
nit (here and elsewhere): `?? fail()` rather than `assertNotUndefined`

Reply:
Oh, why has `assertNotUndefined` fallen out of favor?

Reply:
`x ?? fail(y)` is better than `assertNotUndefined(x, y)` because:

1. It's more readable (you don't have to look within a function call to find x)
2. It's fewer characters
3. It's more efficient (fail function is not evaluated unless x is undefined whereas assertNotUndefined is always evaluated)
4. It's sometimes fewer imports (you have to import fail of course, but fail is also used for other things besides undefined/null checks and might already be imported)

`assertNotUndefined` is still useful in the case where a value might be `null` _or_ `undefined`, and you only want to check for "not undefined" but null is okay. That's a pretty rare situation though.

Reply:
Done